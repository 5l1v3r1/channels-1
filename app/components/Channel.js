// @flow
import React from 'react';
import styles from './Channel.scss';

const Channel = ({
  title,
  thumbnail,
  videoCount,
  subscriberCount,
  isActive,
  updateActiveChannel
}: {
  title: string,
  thumbnail: string,
  videoCount: string,
  subscriberCount: string,
  isActive: boolean,
  updateActiveChannel: (id: string) => void
}) => (
  <li className={isActive ? `${styles.channel} ${styles['channel--isActive']}` : styles.channel} onClick={updateActiveChannel}>
    <img className={styles.thumbnail} src={thumbnail} alt={title} />
    <div>
      <h3 className={styles.title}>{title}</h3>
      <span className={styles.subscribers}>{subscriberCount} subscribers</span>
      <span className={styles.videos}>{videoCount} videos</span>
    </div>
  </li>
);

export default Channel;
