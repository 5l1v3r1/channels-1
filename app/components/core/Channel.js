// @flow
import React from 'react';
import styles from './Channel.scss';

const Channel = ({
  title,
  thumbnail,
  videoCount,
  subscriberCount,
  isActive,
  handleClick
}: {
  title: string,
  thumbnail: string,
  videoCount: string,
  subscriberCount: string,
  isActive: boolean,
  handleClick: (channelId: string) => void
}) => (
  <li className={isActive ? [styles.channel, styles.isActive].join(' ') : styles.channel} onClick={handleClick}>
    <img className={styles.thumbnail} src={thumbnail} alt={title} />
    <div>
      <h3 className={styles.title}>{title}</h3>
      <span className={styles.subscribers}>{subscriberCount} subscribers</span>
      <span className={styles.videos}>{videoCount} videos</span>
    </div>
  </li>
);

export default Channel;
