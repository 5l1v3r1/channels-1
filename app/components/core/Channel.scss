@import '../../app.global.scss';

.channel {
  display: flex;
  align-items: center;
	padding: 0.75rem 1rem;

	&.isActive {
		background-color: $gray;

		.subscribers,
		.videos {
			color: $white;
		}
	}
}

.thumbnail {
  width: 50px;
  height: 50px;
  margin-right: 0.75rem;
  border: 2px solid $white;
  border-radius: 50%;
  box-shadow: 0 1px 3px 1px rgba(46, 46, 46, 0.3);
}

.title {
  margin: 0;

  @include font-size(12);
}

.subscribers,
.videos {
  color: $gray;

  @include font-size(12);
}

.subscribers {
  margin-right: 1rem;
}
