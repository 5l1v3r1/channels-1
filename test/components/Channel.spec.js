import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Channel from '../../app/components/core/Channel';

function setup() {
  const props = {
    title: 'DevTips',
    thumbnail: 'https://yt3.ggpht.com/photo.jpg',
    videoCount: '292',
    subscriberCount: '243,600',
    isActive: false,
    updateActiveChannel: jest.fn()
  };

  const enzymeWrapper = shallow(<Channel {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('Channel', () => {
  it('should render self', () => {
    const { enzymeWrapper } = setup();
    const tree = toJson(enzymeWrapper);

    expect(tree).toMatchSnapshot();
  });

  it('should call updateActiveChannel', () => {
    const { props, enzymeWrapper } = setup();
    const channelId = 'UCfWZwsP8trUy5uHJg8gcGIQ';

    enzymeWrapper.simulate('click', channelId);

    expect(props.updateActiveChannel).toBeCalled();
    expect(props.updateActiveChannel).toBeCalledWith(channelId);
  });
});
