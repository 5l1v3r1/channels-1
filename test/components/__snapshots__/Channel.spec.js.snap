// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel should render self 1`] = `
<li
  className="channel"
  onClick={[Function]}
>
  <img
    alt="DevTips"
    className="thumbnail"
    src="https://yt3.ggpht.com/photo.jpg"
  />
  <div>
    <h3
      className="title"
    >
      DevTips
    </h3>
    <span
      className="subscribers"
    >
      243,600
       subscribers
    </span>
    <span
      className="videos"
    >
      292
       videos
    </span>
  </div>
</li>
`;
